### ISSUE TITLE
Refactor Python package structure

### ISSUE THREAD

**Issue by bfirsh** [embassynetwork/modernomad#349]
Perhaps everything should be under the `modernomad` package instead of being top-level. This will make things a bit more organised, and means we won't collide with other top-level packages.

**Comment by jonathan-s** [comment]
Just a perspective from what it looks like where I work. We've got a folder called apps where all django apps live. If there are several django apps that belong tightly together they would have their own folder. Ie

So perhaps we'd like to have `modernomad` to be our django apps folder?

```
Apps
  - User
      -- Organization
      -- Account
      -- Etc
   - Other app
Some other folder that holds other scripts and stuff
```

**Comment by bfirsh** [comment]
Yeah, I guess so. I think modernomad is all so tightly knit, there is little need to pretend that they are separate re-usable apps. In fact having all bundled inside one Django app has a lot of advantages (like our "core" app is) - simple linear migrations, no import loops, etc.

**Comment by bfirsh** [comment]
The package names `modernomad.core`, `modernomad.api`, etc, are pleasing semantically, too.
